### PR TITLE
Add tags support for Azure File Driver

### PR DESCRIPTION
**Pull request by ZeroMagic**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Add tags support for Azure File Driver.
Ref: https://github.com/kubernetes-sigs/azurefile-csi-driver/issues/281

```
---
kind: StorageClass
apiVersion: storage.k8s.io/v1
metadata:
  name: azurefile-tags
provisioner: kubernetes.io/azure-file
parameters:
  skuname: Standard_LRS
  tags: "key1=val1,key2=val2"
```

Only when creating a new storage account, the tags need to be created. The user's default storage account does not need the tags.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Add tags support for Azure File Driver
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: pkg/volume/azure_file/azure_provision.go [hunk 1]
```diff
@@ -189,6 +193,11 @@ func (a *azureFileProvisioner) Provision(selectedNode *v1.Node, allowedTopologie
 		return nil, fmt.Errorf("claim.Spec.Selector is not supported for dynamic provisioning on Azure file")
 	}
 
+	tags, err := azure_dd.ConvertTagsToMap(customTags)
```
Comment (andyzhangx) [inline pkg/volume/azure_file/azure_provision.go:196]: move this `ConvertTagsToMap` to a common place, e.g. in Azure cloud provider lib?

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go [hunk 2]
```diff
@@ -28,6 +28,12 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// StorageAccount contains the fields which are used to create storage account.
+type StorageAccount struct {
+	Name, Type, Kind, ResourceGroup, Location string
+	Tags                                      map[string]string
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:34]: what about set `map[string]*string` in this struct?

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go [hunk 3]
```diff
@@ -28,6 +28,12 @@ import (
 	"k8s.io/klog/v2"
 )
 
+// StorageAccount contains the fields which are used to create storage account.
+type StorageAccount struct {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:32]: `AccountOptions` is better

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go [hunk 4]
```diff
@@ -35,6 +35,13 @@ type Client struct {
 	fileSharesClient storage.FileSharesClient
 }
 
+// ShareInfo contains the fields which are used to create file share.
+type ShareInfo struct {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go:39]: `ShareOptions` is better

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go [hunk 5]
```diff
@@ -46,27 +53,27 @@ func New(config *azclients.ClientConfig) *Client {
 }
 
 // CreateFileShare creates a file share
-func (c *Client) CreateFileShare(resourceGroupName, accountName, name string, protocol storage.EnabledProtocols, sizeGiB int) error {
-	result, err := c.GetFileShare(resourceGroupName, accountName, name)
+func (c *Client) CreateFileShare(resourceGroupName, accountName string, azureFileShare ShareInfo) error {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go:56]: should pass a pointer of `ShareOptions` in the function

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go [hunk 6]
```diff
@@ -90,7 +96,12 @@ func (az *Cloud) GetStorageAccesskey(account, resourceGroup string) (string, err
 }
 
 // EnsureStorageAccount search storage account, create one storage account(with genAccountNamePrefix) if not found, return accountName, accountKey
-func (az *Cloud) EnsureStorageAccount(accountName, accountType, accountKind, resourceGroup, location, genAccountNamePrefix string, enableHTTPSTrafficOnly bool) (string, string, error) {
+func (az *Cloud) EnsureStorageAccount(account StorageAccount, genAccountNamePrefix string, enableHTTPSTrafficOnly bool) (string, string, error) {
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:99]: should pass a pointer of `AccountOptions` in the function

**Review by andyzhangx — COMMENTED** [review]
use `Options` suffix for Account, Share

File: pkg/volume/azure_file/azure_provision.go (hunk 1 above)
Comment (ZeroMagic) [reply pkg/volume/azure_file/azure_provision.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go (hunk 3 above)
Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go (hunk 4 above)
Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go (hunk 5 above)
Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/clients/fileclient/azure_fileclient.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go (hunk 6 above)
Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go]: done

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storage.go [hunk 7]
```diff
@@ -36,25 +37,32 @@ const (
...
 	}
 
 	enableHTTPSTrafficOnly := true
-	if protocol == storage.NFS {
+	if shareOptions.Protocol == storage.NFS {
 		enableHTTPSTrafficOnly = false
 	}
-	account, key, err := az.EnsureStorageAccount(accountName, accountType, accountKind, resourceGroup, location, fileShareAccountNamePrefix, enableHTTPSTrafficOnly)
+
+	accountName, accountKey, err := az.EnsureStorageAccount(accountOptions, fileShareAccountNamePrefix, enableHTTPSTrafficOnly)
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storage.go:56]: `enableHTTPSTrafficOnly` is also part of `accountOptions` property

Comment (ZeroMagic) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storage.go]: done

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve

**Pull request merged**